gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -0,0 +1,238 @@
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use cln_plugin::{options, Builder, Plugin};
+use cln_rpc::ClnRpc;
+use fedimint_api::{task::TaskGroup, Amount};
+use fedimint_server::config::load_from_file;
+use ln_gateway::{
+    config::ClnRpcConfig,
+    gatewaylnrpc::{
+        gateway_lightning_server::{GatewayLightning, GatewayLightningServer},
+        CompleteHtlcsRequest, CompleteHtlcsResponse, GetPubKeyRequest, GetPubKeyResponse,
+        PayInvoiceRequest, PayInvoiceResponse, SubscribeInterceptHtlcsRequest,
+        SubscribeInterceptHtlcsResponse,
+    },
+};
+use serde::{Deserialize, Deserializer, Serialize};
+use thiserror::Error;
+use tokio::{
+    io::{stdin, stdout},
+    sync::{mpsc, Mutex},
+};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{transport::Server, Status};
+use tracing::error;
+
+// Note: One this binary is stable, we should be able to remove current 'ln_gateway'
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let service = ClnRpcService::new()
+        .await
+        .expect("Failed to create cln rpc service");
+    let dir = service.dir.clone();
+
+    // Read configurations
+    let gw_cfg_path = dir.join("lnrpc.config");
```

Comment:
What do we think about using the `.config` extension for a JSON file? IMO it would be better to just use the standard `.json` extension here and for `gateway.config`.

Also, do we need another config file? Since this just has a bind address can we pass it in via CLI or env?